### PR TITLE
Custom Entity Spawn Data API

### PR DESCRIPTION
**Pull request by TropheusJ**
A small API for sending extra data with entity spawn packets.

Handles making a custom packet for you. Just implement `QuiltCustomSpawnDataEntity` and you can read/write directly to/from a buffer to modify spawn data. Useful when a data tracker is insufficient.

Merging this and #175 into 1 module should be considered.

### CODE REVIEW

File: library/entity/custom_spawn_data/src/main/java/org/quiltmc/qsl/entity/custom_spawn_data/impl/CustomEntitySpawnDataClientInitializer.java
Comment (Leo40Git) [inline library/entity/custom_spawn_data/src/main/java/org/quiltmc/qsl/entity/custom_spawn_data/impl/CustomEntitySpawnDataClientInitializer.java]: `buf` is leaked here!
If you move this `if` into the `try` statement, this will resolve the leak (AFAICR, I _think_ Java runs the `finally` block before returning?)

**Review by Leo40Git — CHANGES_REQUESTED** [review]
oh dear

**Review by LambdAurora — COMMENTED** [review]
I would combine this module and #175's module into a entity_networking module.

I'll take a closer look to this later.

File: library/entity/custom_spawn_data/src/main/java/org/quiltmc/qsl/entity/custom_spawn_data/impl/CustomEntitySpawnDataClientInitializer.java
Comment (SilverAndro) [reply library/entity/custom_spawn_data/src/main/java/org/quiltmc/qsl/entity/custom_spawn_data/impl/CustomEntitySpawnDataClientInitializer.java]: Yes, finally will always run after a try..catch block, essentially "storing" the value to return/throw until after

Comment (TropheusJ) [reply library/entity/custom_spawn_data/src/main/java/org/quiltmc/qsl/entity/custom_spawn_data/impl/CustomEntitySpawnDataClientInitializer.java]: whoops, fixed. Just put the whole thing in the try.

**Comment by TheGlitch76** [comment]
Marking as draft because this PR is marked as `s: not working`; feel free to mark it as ready for review once everything is working again and this is OK to go into QSL :pineapple:

**Comment by TheGlitch76** [comment]
actually--i see that was done by Aurora, not you. I'll leave it to you to determine if this is OK or not.

**Comment by TropheusJ** [comment]
on hold until #175 is merged (to make merging not terrible) and I have time

**Comment by TropheusJ** [comment]
going to resurrect this as a new PR, sorry for letting this sit for so long for no real reason

**Pull request closed without merging**
